public/client/TracyDebug.hpp

```diff
@@ -1,9 +1,25 @@
 #ifndef __TRACYPRINT_HPP__
 #define __TRACYPRINT_HPP__
 
+#ifdef TRACY_ON_DEMAND
+#  define TRACY_VERBOSE_EARLY_OUT_COND if( !GetProfiler().IsConnected() ) break
+#else
+#  define TRACY_VERBOSE_EARLY_OUT_COND assert( tracy::ProfilerAvailable() )
+#endif
+
+#define TracyInternalMessage( severity, ... )																				   \
+	do {																													   \
+        TRACY_VERBOSE_EARLY_OUT_COND;																						   \
+		char buffer[4096];																									   \
+		snprintf( buffer, sizeof(buffer), __VA_ARGS__ );																	   \
+		tracy::Profiler::LogString( tracy::MessageSourceType::Tracy, severity, 0, TRACY_CALLSTACK, strlen( buffer ), buffer ); \
+	} while( 0 )
+
 #ifdef TRACY_VERBOSE
 #  include <stdio.h>
-#  define TracyDebug(...) fprintf( stderr, __VA_ARGS__ );
+#  define TracyDebug(...) do { fprintf( stderr, __VA_ARGS__ ); fputc( '\n' ); } while( 0 )
```

Comment:
Is there `stderr` missing in `fputc` call? The man page states the signature for the function is `int fputc(int c, FILE *stream);`.

Reply:
indeed an error on my part